src/domain/dataset-update-flow/src/entities/update/update_event.rs

```diff
@@ -0,0 +1,153 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use chrono::{DateTime, Utc};
+use enum_variants::*;
+use kamu_task_system::TaskID;
+use opendatafabric::DatasetID;
+
+use crate::*;
+
+/////////////////////////////////////////////////////////////////////////////////////////
+
+/// All events that model life-cycle of a task
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum UpdateEvent {
+    /// Update initiated
+    Initiated(UpdateInitiated),
+    /// Postponed
+    Postponed(UpdatePostponed),
+    /// Queued for time
+    Queued(UpdateQueued),
+    /// Secondary triger
+    SecondaryTrigger(UpdateSecondaryTrigger),
+    /// Scheduled/Rescheduled a task
+    TaskScheduled(UpdateTaskScheduled),
+    /// Succeeded task
+    TaskSucceeded(UpdateTaskSucceeded),
+    /// Failed task
+    TaskFailed(UpdateTaskFailed),
```

Comment:
In the `Task` model I found it very nice to separate the fact that task **finished** from its **outcome** (success/failure/cancellation). I suggest doing the same here.

Reply:
Compacted